android ip cam sensor: update battery icon

happyleavesaoc opened the pull request:
## Description:

Two changes to the android ip cam battery sensor:

* use rounded level to apply icon (bugfix)
* use alert icon instead of outline icon (subjective)

## Checklist:

If the code communicates with devices, web services, or third-party tools:
  - [x] Local tests with `tox` run successfully. **Your PR cannot be merged unless tests pass**

[ex-requir]: https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/keyboard.py#L14
[ex-import]: https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/keyboard.py#L54

balloob committed:
Version bump to 0.54

MisterWil committed:
Update AbodePy to 0.11.8 (#9537)

* Update requirements_all.txt

* Update abode.py

arsaboo committed:
Bump python_openzwave to 0.4.0.35 (#9542)

* Bump python_openzwave to 0.4.0.35

* Cleanup

happyleavesaoc committed:
update usps (#9540)

* update usps

* fix syntax issue

balloob committed:
Merge pull request #9532 from home-assistant/release-0-54

0.54

happyleavesaoc committed:
update battery icon

happyleavesaoc committed:
update battery icon

balloob commented on homeassistant/components/sensor/android_ip_webcam.py:
```diff
@@ -78,8 +78,8 @@ def icon(self):
             if rounded_level < 10:
-                returning_icon = 'mdi:battery-outline'
-            elif self._state == 100:
+                returning_icon = 'mdi:battery-alert'
```
Can you rewrite this to use [`icon_for_battery_level`](https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/util/icon.py#L5). And then update that method to use `battery-alert`.

Also, that method needs to be moved to `helpers` because it is Home Assistant specific.

I just left a similar comment about moving it to helpers so maybe conflicts incoming https://github.com/home-assistant/home-assistant/pull/9594#discussion_r141249519

MartinHjelmare replied:
How is the function specific to home assistant?

balloob replied:
Icons are using the Material Design Icon set, which is what we use. All other things in the util package would be useful for any Python project.

That being said, Pascal just told me he was confused too. So maybe I'm wrong here

happyleavesaoc replied:
Didn't know that other method existed! Thanks. I'll wait for a consensus on method location.

MartinHjelmare replied:
Ok, fair enough. At least the function is specific for all projects using material design icons. Moving it to helpers is probably correct.

balloob replied:
Alright, let's move it. Can you do it @happyleavesaoc ?

happyleavesaoc replied:
@balloob Yep I can do it. ETA this weekend.

tchellomello replied:
@happyleavesaoc  once you move it, please let me know so then I can edit the PR #9594 as well. Thank you! :+1:

Commit:
New Wink services. pair new device, rename, and delete, add new lock key code. Add water heater support (#9303)

* Pair new device, rename, delete, and lock key code services. Also add water heater support.

* Fixed tox

bachya committed:
Fixes UPS MyChoice exception (#9587)

* Fixes UPS MyChoice exception

* Added unit of measurement

* Collaborator-requested changes

bachya committed:
FedEx: Adds "packages" as a unit (#9588)

* Adds "packages" as a unit

* Collaborator-requested changes

pvizeli committed:
Cleanup entity & remove warning (#9606)

* Cleanup entity & remove warning

* Update comment

balloob committed:
Add OwnTracks over HTTP (#9582)

* Add OwnTracks over HTTP

* Fix tests

nkgilley committed:
upgrade python-ecobee-api (#9612)

djchen committed:
Change TP-Link Switch power statistics attribute names (#9607)

balloob committed:
Migrate Alexa smart home to registry (#9616)

* Migrate Alexa smart home to registry

* Fix tests

tinloaf committed:
Add an input_datetime (#9313)

* Initial proposal for the input_datetime

* Linting

* Further linting, don't define time validation twice

* Make pylint *and* flake8 happy at the same time

* Move todos to the PR to make lint happy

* Actually validate the type of date/time

* First testing

* Linting

* Address code review issues

* Code review: Remove forgotten print()s

* Make set_datetime a coroutine

* Create contains_at_least_one_key_value CV method, use it

* Add timestamp to the attributes

* Test and fix corner case where restore data is bogus

* Add FIXME

* Fix date/time setting

* Fix Validation

* Merge date / time validation, add tests

* Simplify service data validation

* No default for initial state, allow 'unknown' as state

* cleanup

* fix schema

Commit:
Cover component for RFlink (#9432)

* second try on rflink / cover

* no newline at end of file

* changed entity

* fixed comments from pvizeli

* removed :

* removed return 'unknown'

* Fixed comments from Rytilahti

* removed newline

* Reverted to None

* cleanup

* Cleanup

tchellomello committed:
Introducing support to Melnor RainCloud sprinkler systems (#9287)

*  Introducing support to Melnor RainCloud sprinkler systems

* Make monitored_conditions optional for sub-components

*  Part 1/2 - Modified attributes, added DATA_ constant and using battery helper

* Part 2/2 - Refactored self-update hub

* Fixed change requested:
- Dispatcher signal connection
- Don't send raincloud object via dispatcher_send()
- Honoring the dynamic scan_interval value on track_time_interval()

* Inherents async_added_to_hass() on all device classes

* Makes lint happy

* * Refactored RainCloud code to incorporate suggestions.
  Many thanks to @pvizelli and @martinhjelmare!!

* Removed Entity from RainCloud sensor and fixed docstrings

* Update raincloud.py

* Update raincloud.py

* fix lint

snjoetw committed:
Rewrite synology camera by using py-synology package (#9583)

* - Rewrite synology camera by intruducing Api and SurveillanceStation classes to get cameras, motion settings, enable/disable motion detection, etc ...
- Synology camera now shows correct state based on is_recording and is_streaming flag. Also it now supports enable / disable motion detection and show the correct motion detection status
- Newly added Api and SurveillanceStation classes will be moved to a lib but it's here just for review

* - Updated how payload are merged with kwargs so it works with python <3.5

* - Fixed class name conflict

* - Addressed flake8 error

* - Addressed pylint error

* - Moved synology API related code to py-synology lib
- Added py-synology==0.1.1 requirement
- Removed hass from SynologyCamera constructor

* - Updated requirements_all.txt

* - renamed variable back to original

* - Sync call to retrieve camera image should be done in camera_image() instead

* - Sync call to update camera info should be done in update() instead

* - Removed unused import

rytilahti committed:
yeelight: allow turn_off transitions, fixes #9602 (#9605)

fabaff committed:
Move 'voltage' to const (#9621)

jalmeroth committed:
Yamaha MusicCast: check known_hosts (#9580)

* Yamaha MusicCast: check known_hosts

- pymusiccast: Version bump

* Update requirements

etsinko committed:
Fixed away_mode for Ecobee thermostat. (#9559)

* Fixed away_mode for Ecobee thermostat. Now away mode is properly turned on using indefinite away hold.

* fixed lint warnings

* fixed lint warnings

* - now it is possible to use float values for ecobee temperature holds
- fixed a bug that caused an exception when temperature hold was set in away mode
- added unit tests for ecobee thermostat

* fixed lint errors

* fixed lint errors

alanfischer committed:
Switched VeraSensor to use category ids (#9624)

philk committed:
splunk: Handle datetime objects in event payload (#9628)

If an event contained a datetime.datetime object it would cause an
exception in the Splunk component. Most of the media_player
components do this in their `media_position_updated_at` attribute.

Use the JSONEncoder from homeassistant.remote instead of just using the
standard json.dumps encoder.

Fixes #9590

tinloaf committed:
MQTT climate platform [continuation of #8750] (#9589)

* New climate platform with MQTT

* Use STATE_OFF

* Basic tests for climate.mqtt

* lint

* actualy collect coverage

* First tests and fixes

* Add possibility to receive temperature via MQTT

* Require only either sensor or mqtt topic

* Add mqtt publishing for away mode, hold mode and aux heat.

* Use configurabe on/off payloads

* Add pessimistic mode

* Initialize aux and away with False instead of None

* Remove Sensor

* Use correct scheduling method

* Move all methods to coroutines

szogi committed:
wunderground: fix supported language codes #9631 (#9633)

* removed PU, added TR language code (https://www.wunderground.com/weather/api/d/docs?d=language-support&MR=1), fixes #9631

rytilahti committed:
add myself to codeowners (#9642)

szogi committed:
raspihats: unmet dependency fix (#9638)

* raspihats: update to 2.2.3 (deps fix)

Raspihats platform update, upstream fixed enum34 requirements, added smbus dependency
Fixes #9547

* raspihats: update to 2.2.3, smbus-cffi dependency

Raspihats platform update, upstream fixed enum34 requirements, added smbus dependency
Fixes #9547

* raspihats: update to 2.2.3

* raspihats: update to 2.2.3, smbus-cffi dependency

* raspihats: update to 2.2.3, smbus-cffi dependency

* raspihats: update to 2.2.3 (deps fix)

Raspihats platform update, upstream fixed enum34 requirements, added smbus dependency
Fixes #9547

* raspihats: update to 2.2.3, smbus-cffi dependency

David-Byrne committed:
Fixes broken source links in API docs (#9636)

* Fixes broken source links in API docs

* Removes illegal blank line

balloob committed:
Move group services into their own YAML (#9597)

* Move group services into their own YAML

* Fix lint

* Move persistent notification to package

mweimerskirch committed:
Facebook Messenger notify component: add support for sending messages to specific page user IDs (#9643)

vickyg3 committed:
arlo: Add battery level sensor (#9637)

* arlo: Add battery level sensor

Adds a battery level sensor that monitors the battery level on
Arlo cameras.

* Fix lint issue

hotplot committed:
Add hysteresis attribute to threshold binary sensor (#9596)

* Added hysteresis attribute to threshold binary sensor

* Added threshold binary sensor hysteresis test case

* Changed threshold binary sensor property name to be more self explanatory

* Pulled default hysteresis value into top level declaration

* Fixed linter errors

* Fixed additional linter errors

* Move comment to docs

fabaff committed:
Upgrade numpy to 1.13.3 (#9646)

fabaff committed:
Upgrade youtube_dl to 2017.10.01 (#9647)

fabaff committed:
Upgrade discord.py to 0.16.12 (#9648)

fabaff committed:
Upgrade netdisco to 1.2.2 (#9649)

fabaff committed:
Upgrade influxdb to 4.1.1 (#9652)

* Upgrade influxdb to 4.1.1

* Upgrade influxdb to 4.1.1

fabaff committed:
Upgrade googlemaps to 2.5.1 (#9653)

mw-white committed:
mqtt_statestream: Add options to publish attributes/timestamps (#9645)

jeroenterheerdt committed:
Fixed bugs related to exception handling in pythonegardia. Updating package requirement accordingly (#9663)

arsaboo committed:
Update google-api-python-client to 1.6.4 (#9658)

MisterWil committed:
Bump abode to 0.11.9 (#9660)

emlove committed:
Unit tests to improve core coverage (#9659)

* Code coverage of logging util

* Improve async util coverage

* Add test coverage for restore_state

* get_random_string test

alanfischer committed:
Upgrade pyitachip2ir to 0.0.7 (#9669)

FletcherAU committed:
Fix typo in cancel_command description (#9671)

"wasn't going to use it"

BioSehnsucht committed:
Rename input_slider to input_number and add numeric text box option (#9494)

* * Rename input_slider to input_number
* Update input_number to optionally display slider, input box, or both

* input_number support either input box or slider mode, but not both

* input_number : change service from select_value to set_value

* input_number : add test for mode setting to tests

alanfischer committed:
Properly handle an invalid end_time (#9675)

pvizeli committed:
Support new feature to push API data to hassio (#9679)

* Support new featuer to push API data to hassio

* Add tests & services

jeroenterheerdt committed:
Adding ignore capability to Egardia component (#9676)

mbrrg committed:
Fix Google Calendar/oauth2client warning (#9677)

* Fixes oauth2client warning.

* Fix permission.

aequitas committed:
Implement DSMR5 support. (#9686)

* Allow configuring DSMR5 protocol.

* Give good example.

* Using dev branch until released upstream.

* Update to dsmr_parser supporting v5 arguments.

* Update to latest dmsr parser, preventing exceptions thrown where warnings would suffice.

* Update even more

* Update requirements.

* Update requirements

Danielhiversen committed:
Add Tibber sensor (#9661)

* Add Tibber sensor

* remove extra space

milanvo committed:
Add recorder purge service, rework purge timer (#9523)

* Add recorder purge service

* Recorder test to match purge config

* Removed purge timer, move service handler to setup, add service description file

* Tests for recorder purge service

* Recorder purge timer rework, add purge service parameter, tests

* Purge service schema change

* Service description change value range

* First cleanup

* Fix name of config

milanvo committed:
Fix restore state by filter out null value row from DB query (#9690)

jeroenterheerdt committed:
Updating clicksendaudio component based on feedback (#9692)

* Updating clicksendaudio component based on feedback

* Updating .coveragerc - forgot to add new file clicksendaudio.py

bestlibre committed:
Refactoring of onewire sensor component (#9691)

pvizeli committed:
HassIO replace config changes (#9695)

* Update flow

* fix tests

* Update hassio.py

arsaboo committed:
Fixed typo in opencv (#9697)

lwis committed:
[light.tradfri] async support with resource observation. (#7815)

* [light.tradfri] Initial support for observe

* Update for pytradfri 2.0

* Fix imports

* Fix missing call

* Don't yield from add devices

* Fix imports

* Minor fixes to async code.

* Imports, formatting

* Docker updates, some minor async code changes.

* Lint

* Lint

* Update pytradfri

* Minor updates for release version

* Build fixes

* Retry observation if failed

* Revert

* Additional logging, fix returns

* Fix rename

* Bump version

* Bump version

* Support transitions

* Lint

* Fix transitions

* Update Dockerfile

* Set temp first

* Observation error handling

* Lint

* Lint

* Lint

* Merge upstream changes

* Fix bugs

* Fix bugs

* Fix bugs

* Lint

* Add sensor

* Add sensor

* Move sensor attrs

* Filter devices better

* Lint

* Address comments

* Pin aiocoap

* Fix bug if no devices

* Requirements

balloob committed:
RFC: Create a secrets file and enable HTTP password by default (#9685)

* Create a secret and enable password by default

* Comment out api password secret

* Lint/fix tests

fabaff committed:
Support for The Things Network (#9627)

* Support for The Things network's Data Storage

* Rename platform and other changes (async and dict)

* Rename sensor platform and remove check for 200

balloob committed:
Update frontend

balloob committed:
Merge remote-tracking branch 'origin/master' into dev

flowolf committed:
fixed duplicate words (#9705)

happyleavesaoc committed:
move icon battery function from util to helpers

happyleavesaoc committed:
remerge
